doc/content/components/network-server.md

```diff
@@ -7,3 +7,54 @@ weight: 3
 The Network Server handles the LoRaWAN network layer, including MAC commands, regional parameters and adaptive data rate (ADR).
 
 <!--more-->
+
+## Device Management
+
+Network Servers expose [NsEndDeviceRegistry]({{< ref "/reference/api/end_device#the-nsenddeviceregistry-service" >}}) service for end device management. Typical clients of this sevice are [Console]({{< ref "/components/console.md" >}}) and [CLI]({{< ref "/components/cli.md" >}}).
+
+Network Servers store device MAC configuration, MAC state and network session keys.
+
+Change of device MAC configuration may trigger a downlink message.
+
+## Application Downlink Queue Management and Linking
+
+Network Servers expose [AsNs]({{< ref "/reference/api/network_server#the-asns-service" >}}) service for application downlink management and linking. Typical clients of this service are [Application Servers]({{< ref "/components/application-server.md" >}}).
+
+Change of application downlink queue may trigger a downlink message.
+
+Once the link is established, Network Server will send application-specific uplink messages to the client via the link. There can be at most one active link per-appplication.
+
+In case link is not active, but Network Server has application-specific uplink messages to send, those messages will be queued and sent once the link is established.
+
+## Downlink Scheduling
+
+Network Server maintains internal downlink task queue. Each downlink task has an execution time associated with it, which tasks are sorted by in ascending order. Whenever a downlink task is ready to execute, it is executed as soon as possible.
```

Comment:
I find this too much behavior description and not what NS is, but that is my opinion. No need for change, but let's be aware of a sliding edge that we cannot exhaustively describe all our component's behavior, and nobody really cares